assets/src/editor/events/initialization/editor-initialization.js

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+
+import useEditorInitialization from './use-editor-initialization';
+import withEventEntity from './with-event-entity';
+import withEventDateEntities from './with-event-date-entities';
+import withTicketEntitiesForAllDateEntities from './with-ticket-entities-for-all-date-entities';
+import { Provider } from './context';
+
+const EditorInitialization = ( {
+	eventEntityLoaded,
+	dateEntitiesLoaded,
+	ticketEntitiesLoaded,
+	editorInitialized,
+	setInitialized,
+	children,
+} ) => {
+	useEffect( () => {
+		if ( eventEntityLoaded && dateEntitiesLoaded && ticketEntitiesLoaded ) {
+			setInitialized( true );
+		}
+	}, [ eventEntityLoaded, dateEntitiesLoaded, ticketEntitiesLoaded ] );
+	return <Provider value={ editorInitialized }>
+		{ children }
+	</Provider>;
+};
+
+export default compose( [
+	( WrappedComponent ) => ( props ) => {
+		const [ editorInitialized, setInitialized ] = useEditorInitialization();
```

Comment:
If it was going to be exported standalone yeah maybe, however because this is exported as part of a composite, for the purpose of debugging the `EditorInitialization` name will be present.  So unnecessary in this case.